### PR TITLE
First version of aero visual localization pipeline

### PR DESCRIPTION
**Pull request by vnmsklnk**
This PR includes the VPR systems AnyLoc, CosPlace, EigenPlaces, MixVPR, NetVLAD; LightGlue and SuperGlue key point matchers. Evaluation of results is possible with the added Recall metric. Also added functionality to download satellite map from Google Maps in tile format.

### CODE REVIEW

File: aero_vloc/feature_matchers/feature_matcher.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#  limitations under the License.
+import torch
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+
+class FeatureMatcher(ABC):
+    def __init__(self, resize: int):
+        self.resize = resize
```
Comment (anastasiia-kornilova) [inline aero_vloc/feature_matchers/feature_matcher.py:22]: `scale_factor`, maybe?

File: aero_vloc/localization_pipeline/localization_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        self.sat_map = sat_map
+
+        global_descs = []
+        for image in tqdm(
+            sat_map, desc="Calculating of global descriptors for source DB"
+        ):
+            global_descs.append(self.vpr_system.get_image_descriptor(image.path))
+        self.source_global_descs = np.asarray(global_descs)
+
+        self.faiss_index = faiss.IndexFlatL2(self.source_global_descs.shape[1])
```
Comment (anastasiia-kornilova) [inline aero_vloc/localization_pipeline/localization_pipeline.py:51]: Strategy of index search to be moved to separate class, I think

File: aero_vloc/localization_pipeline/localization_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            print("Perspective transform error. Abort matching")
+            return None
+
+        moments = cv2.moments(dst)
+        cX = int(moments["m10"] / moments["m00"])
+        cY = int(moments["m01"] / moments["m00"])
+
+        center = (cX / self.feature_matcher.resize, cY / self.feature_matcher.resize)
+
+        latitude = chosen_sat_image.top_left_lat + abs(center[1]) * (
```
Comment (anastasiia-kornilova) [inline aero_vloc/localization_pipeline/localization_pipeline.py:156]: Geo-referencing to be moved to separate class later, let's add issue
That will be Anna part

Comment (anastasiia-kornilova) [reply aero_vloc/localization_pipeline/localization_pipeline.py:156]: Also, reprojection error reporting

File: aero_vloc/metrics/recall.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    :return: Recall value
+    """
+    positives = 0
+    for drone_image in drone_seq:
+        res = localization_pipeline.localize(drone_image, k_closest, visualize=False)
+        if res is not None:
+            lat, lon, _ = res
+            error = calculate_distance(
+                lat, lon, drone_image.gt_latitude, drone_image.gt_longitude
```
Comment (anastasiia-kornilova) [inline aero_vloc/metrics/recall.py:43]: It will be correct only for nadir images, calculated long/lat are on ground (center of image), drone gt is in air, I think

Comment (anastasiia-kornilova) [reply aero_vloc/metrics/recall.py:43]: Let's add issue to improve this calculation

**Review by anastasiia-kornilova — CHANGES_REQUESTED** [review]

File: aero_vloc/localization_pipeline/localization_pipeline.py (hunk 3 above)
Comment (vnmsklnk) [reply aero_vloc/localization_pipeline/localization_pipeline.py:152]: #2

File: aero_vloc/metrics/recall.py (hunk 4 above)
Comment (vnmsklnk) [reply aero_vloc/metrics/recall.py:43]: #3

**Review by anastasiia-kornilova — APPROVED** [review]
